### DISCUSSION TITLE
Ubuntu 22.04 guest suddenly networking not available.

### DISCUSSION THREAD

**Discussion by xlorepdarkhelm** [utmapp/UTM#4832] [UTM Help (macOS)]
**BEFORE SUBMITTING YOUR ISSUE, PLEASE LOOK AT THE PINNED ISSUES AND USE THE SEARCH FUNCTION TO MAKE SURE IT IS NOT ALREADY REPORTED. ALWAYS COMMENT ON AN EXISTING ISSUE INSTEAD OF MAKING A NEW ONE.**

**Describe the issue**  

I try logging into the guest, and have no network connectivity.  The device is visible in the guest through lspci and lshw, but shown as DISABLED. Nothing seems to be enabling it.

I have tried the Shared and Bridged network with no success on either.

**Configuration**  
* UTM Version: 4.0.9
* macOS Version: 12.62 Monterey
* Mac Chip (Intel, M1, ...): M1

**Crash log**  

N/A

**Debug log**  

Running:  -L /Applications/UTM.app/Contents/Resources/qemu -S -spice "unix=on,addr=/Users/chill/Library/Group Containers/WDNLXAD4W8.com.utmapp.UTM/34D3D11F-DDEE-4807-9EA7-82404AAC6F32.spice,disable-ticketing=on,image-compression=off,playback-compression=off,streaming-video=off,gl=off" -chardev spiceport,id=org.qemu.monitor.qmp,name=org.qemu.monitor.qmp.0 -mon chardev=org.qemu.monitor.qmp,mode=control -nodefaults -vga none -device virtio-net-pci,mac=E2:8E:00:F1:EB:BA,netdev=net0 -netdev vmnet-bridged,id=net0,ifname=en0 -device virtio-ramfb -cpu host -smp cpus=4,sockets=1,cores=4,threads=1 -machine virt -accel hvf -drive if=pflash,format=raw,unit=0,file=/Applications/UTM.app/Contents/Resources/qemu/edk2-aarch64-code.fd,readonly=on -drive if=pflash,unit=1,file=/Users/chill/Library/Containers/com.utmapp.UTM/Data/Documents/Nibenay.utm/Data/efi_vars.fd -m 8192 -device intel-hda -device hda-duplex -device nec-usb-xhci,id=usb-bus -device usb-tablet,bus=usb-bus.0 -device usb-mouse,bus=usb-bus.0 -device usb-kbd,bus=usb-bus.0 -device qemu-xhci,id=usb-controller-0 -chardev spicevmc,name=usbredir,id=usbredirchardev0 -device usb-redir,chardev=usbredirchardev0,id=usbredirdev0,bus=usb-controller-0.0 -chardev spicevmc,name=usbredir,id=usbredirchardev1 -device usb-redir,chardev=usbredirchardev1,id=usbredirdev1,bus=usb-controller-0.0 -chardev spicevmc,name=usbredir,id=usbredirchardev2 -device usb-redir,chardev=usbredirchardev2,id=usbredirdev2,bus=usb-controller-0.0 -device usb-storage,drive=drivecdrom0,removable=true,bootindex=0,bus=usb-bus.0 -drive if=none,media=cdrom,id=drivecdrom0 -device virtio-blk-pci,drive=drive0,bootindex=1 -drive if=none,media=disk,id=drive0,file=/Users/chill/Library/Containers/com.utmapp.UTM/Data/Documents/Nibenay.utm/Data/data.qcow2,discard=unmap,detect-zeroes=unmap -device virtio-serial -device virtserialport,chardev=vdagent,name=com.redhat.spice.0 -chardev spicevmc,id=vdagent,debug=0,name=vdagent -fsdev local,id=virtfs0,path=/Users/chill/Documents/Shared-VM,security_model=mapped-xattr -device virtio-9p-pci,fsdev=virtfs0,mount_tag=share -name Nibenay -uuid 34D3D11F-DDEE-4807-9EA7-82404AAC6F32 -rtc base=localtime -device virtio-rng-pci
qemu-aarch64-softmmu: warning: Spice: playback:0 (0x1438c6140): setsockopt failed, Operation not supported on socket
qemu-aarch64-softmmu: warning: Spice: record:0 (0x1438c61e0): setsockopt failed, Operation not supported on socket

**Upload VM**  

[config.plist.zip](https://github.com/utmapp/UTM/files/10270982/config.plist.zip)

**Comment by ktprograms** [comment]
What is the name of the network interface? Is it the same as the one in `/etc/netplan/00-installer-config.yaml`?

**Comment by xlorepdarkhelm** [comment]
```
$ lspci -vvvv
...

00:01.0 Ethernet controller: Red Hat, Inc. Virtio network device
        Subsystem: Red Hat, Inc. Virtio network device
        Control: I/O+ Mem+ BusMaster+ SpecCycle- MemWINV- VGASnoop- ParErr- Stepping- SERR- FastB2B- DisINTx+
        Status: Cap+ 66MHz- UDF- FastB2B- ParErr- DEVSEL=fast >TAbort- <TAbort- <MAbort- >SERR- <PERR- INTx-
        Latency: 0
        Interrupt: pin A routed to IRQ 46
        Region 0: I/O ports at 10c0 [size=32]
        Region 1: Memory at 10064000 (32-bit, non-prefetchable) [size=4K]
        Region 4: Memory at 10040000 (64-bit, prefetchable) [size=16K]
        Expansion ROM at 10000000 [disabled] [size=256K]
        Capabilities: <access denied>
        Kernel driver in use: virtio-pci

...
```

```
$ sudo lshw -c net
  *-network
       description: Ethernet controller
       product: Virtio network device
       vendor: Red Hat, Inc.
       physical id: 1
       bus info: pci@0000:00:01.0
       version: 00
       width: 64 bits
       clock: 33MHz
       capabilities: msix bus_master cap_list rom
       configuration: driver=virtio-pci latency=0
       resources: irq:46 ioport:10c0(size=32) memory:10064000-10064fff memory:10040000-10043fff memory:10000000-1003ffff
     *-virtio0 DISABLED
          description: Ethernet interface
          physical id: 0
          bus info: virtio@0
          logical name: enp0s1
          serial: e2:8e:00:f1:eb:ba
          capabilities: ethernet physical
          configuration: autonegotiation=off broadcast=yes driver=virtio_net driverversion=1.0.0 link=no multicast=yes
```

and

```
$ cat /etc/netplan/00-installer-config.yaml
# This is the network config written by 'subiquity'
network:
  ethernets:
    enp0s6:
      dhcp4: true
  version: 2
```

**Comment by ktprograms** [comment]
The interface name in the netplan config isn't the same as the actual name of the network device. Change it to enp0s1 and it should work.

Out of curiosity, did this happen after upgrading to UTM 4.x? If so, then its because of the change described here: https://docs.getutm.app/updates/v4.0/#notes (at the last bullet point)

**Comment by xlorepdarkhelm** [comment]
That did it, thanks. Missed that change.

**Comment by pjboud** [comment]
I have the same issue, but in my case the interface name is set to enp0s1.

**Configuration:**

- UTM 4.1.5
- VM: Ubuntu 22.04
- MacOS Ventura 13.1
- Mac M1 chip

---------------
```
~$ sudo lshw -c net
*-network                 
       description: Ethernet controller
       product: Virtio network device
       vendor: Red Hat, Inc.
       physical id: 1
       bus info: pci@0000:00:01.0
       logical name: /dev/fb0
       version: 00
       width: 64 bits
       clock: 33MHz
       capabilities: msix bus_master cap_list rom fb
       configuration: depth=32 driver=virtio-pci latency=0 mode=1280x800 visual=truecolor xres=1280 yres=800
       resources: iomemory:800-7ff irq:46 ioport:10c0(size=32) memory:10044000-10044fff memory:8000000000-8000003fff memory:10000000-1003ffff
     *-virtio0
          description: Ethernet interface
          physical id: 0
          bus info: virtio@0
          logical name: enp0s1
          serial: 5a:15:fe:31:2e:30
          capabilities: ethernet physical
          configuration: autonegotiation=off broadcast=yes driver=virtio_net driverversion=1.0.0 ip='removed by myself' link=yes multicast=yes

cat /etc/netplan/00-installer-config.yaml
network:
  ethernets:
    enp0s1:
      dhcp4: true
  version: 2
```

Set to Shared, but also tried tried Bridged with no success.
